conformance_tests/test_tiovx/test_bidir_ext_standardnode.c

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2013-2023 The Khronos Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "test_engine/test.h"
+#include "test_tiovx/test_tiovx.h"
+
+#include <VX/vx.h>
+#include <TI/tivx.h>
+#include <VX/vx_khr_bidirectional_parameters.h>
+
+TESTCASE(bpExtStandardNodes, CT_VXContext, ct_setup_vx_context, 0)
+
+/* Write some data on an 8-bit image */
+void bd_std_node_writeImage(vx_image image, vx_uint8 a, vx_uint8 b)
+{
+    vx_map_id id;
+    vx_rectangle_t rect = {.start_x = 0, .end_x = 1, .start_y = 0, .end_y = 2};
+    vx_imagepatch_addressing_t ipa;
+    void *ptr;
+    VX_CALL(vxMapImagePatch(image, &rect, 0, &id, &ipa, &ptr, VX_READ_AND_WRITE, VX_MEMORY_TYPE_HOST, 0));
+    *(vx_uint8 *)vxFormatImagePatchAddress2d(ptr, 0, 0, &ipa) = a;
+    *(vx_uint8 *)vxFormatImagePatchAddress2d(ptr, 0, 1, &ipa) = b;
+    VX_CALL(vxUnmapImagePatch(image, id));
+}
+
+/* Check data in an 8-bit image */
+vx_status bd_std_node_checkImage(vx_image image, vx_uint8 a, vx_uint8 b)
+{
+    vx_status status = (vx_status)VX_SUCCESS;
+    vx_map_id id;
+    vx_rectangle_t rect = {.start_x = 0, .end_x = 1, .start_y = 0, .end_y = 2};
+    vx_imagepatch_addressing_t ipa;
+    void *ptr;
+    status = vxMapImagePatch(image, &rect, 0, &id, &ipa, &ptr, VX_READ_AND_WRITE, VX_MEMORY_TYPE_HOST, 0);
```

Comment:
status is returned, but not checked.  Better to use VX_CALL() which checks the status and asserts error if found.  This is found several times in this file.  Please correct all.

Reply:
you can't use the VX_CALL as I explained you in my previous post. The VX_CALL can return a void.
the status is checked later in the code when we use the function

Reply:
The point is that you set status in line 47, but overwrite it in line 50 before checking it.  What if line 47 fails but line 50 passes ... you miss the fail when you check.

Reply:
Yes, this is fine with me